financial-templates-lib/test/GasEstimator.js

```diff
@@ -0,0 +1,57 @@
+const { delay } = require('../delay');
+
+const { GasEstimator } = require("../GasEstimator");
+
+contract("GasEstimator.js", function() {
+
+  let gasEstimator;
+
+
+  describe("Construction with default config", () => {
+    beforeEach(() => {
+        gasEstimator = new GasEstimator();
+    })
+
+    it("Default parameters are set correctly", () => {
+        assert(gasEstimator.updateThreshold > 0);
+        assert(gasEstimator.defaultFastPriceGwei > 0);
+    });
+    it("Returns gas prices in wei before initial update", () => {
+        assert.equal(gasEstimator.defaultFastPriceGwei, gasEstimator.getCurrentFastPrice()/1e9);
+    });
+    it("Returns gas prices in wei after update", async () => {
+        await gasEstimator.update();
+        const latestFastGasPrice = gasEstimator.getCurrentFastPrice() / 1e9;
+        if (latestFastGasPrice === gasEstimator.defaultFastPriceGwei) {
+            console.log(`API Request to ethgasstation.info failed, using default gas price in Gwei: ${latestFastGasPrice}`);
+        } 
+        assert(latestFastGasPrice > 0);
```

Comment:
Do you want to call the API directly from the test script and validate the output? might be redundant though but this test does not actually check the price changes from the `defaultFastPriceGwei`.

Not sure, it's hard to test due to the coupling with `ethGasStation`.

Reply:
So I thought about that but what if the API returns the same result as `defaultFastPriceGwei`?